packages/client/src/object/getLinkedObjectByPkOrThrow.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ObjectTypeKeysFrom, OntologyDefinition } from "@osdk/api";
+import { getLinkedObjectV2 } from "@osdk/gateway/requests";
+import type { ClientContext } from "@osdk/shared.net";
+import { createOpenApiRequest } from "@osdk/shared.net";
+import { convertWireToOsdkObjects } from "./convertWireToOsdkObjects.js";
+
+export async function getLinkedObjectByPkOrThrow<
+  O extends OntologyDefinition<any>,
+  T extends ObjectTypeKeysFrom<O> & string,
+>(
+  client: ClientContext<O>,
+  sourceApiName: T,
+  primaryKey: any,
+  linkTypeApiName: string,
+  linkedObjectPrimaryKey: any,
+) {
+  const object = await getLinkedObjectV2(
+    createOpenApiRequest(client.stack, client.fetch),
+    client.ontology.metadata.ontologyApiName,
+    sourceApiName,
+    primaryKey,
+    linkTypeApiName,
+    linkedObjectPrimaryKey,
+    {
+      select: [],
+    },
+  );
+
+  return convertWireToOsdkObjects(
+    client,
+    client.ontology.objects[sourceApiName].links[linkTypeApiName].targetType,
```

Comment:
Do we actually want to be passing in apiName here? Or just grab `__apiName` from each object we convert?

Reply:
I am guessing that its equiv for now. In fact you could add an invariant check here until interface support that says that we expect them to be the same

Reply:
(OR you stop having to look up the API name entirely, which also matches the legacy-client behavior)